packages/x-charts/src/context/CartesianProvider/Cartesian.types.ts

```diff
@@ -0,0 +1,70 @@
+import { DatasetType } from '../../models/seriesType/config';
+import {
+  AxisDefaultized,
+  ScaleName,
+  ChartsXAxisProps,
+  ChartsYAxisProps,
+  AxisId,
+  AxisConfig,
+} from '../../models/axis';
+
+export type CartesianProviderProps = {
+  /**
+   * The configuration of the x-axes.
+   * If not provided, a default axis config is used.
+   * An array of [[AxisConfig]] objects.
+   */
+  xAxis: AxisConfig<ScaleName, any, ChartsXAxisProps>[];
+  /**
+   * The configuration of the y-axes.
+   * If not provided, a default axis config is used.
+   * An array of [[AxisConfig]] objects.
+   */
+  yAxis: AxisConfig<ScaleName, any, ChartsYAxisProps>[];
+  /**
+   * An array of objects that can be used to populate series and axes data using their `dataKey` property.
+   */
+  dataset?: DatasetType;
+  children: React.ReactNode;
+};
+
+export type DefaultizedAxisConfig<AxisProps> = {
+  [axisId: AxisId]: AxisDefaultized<ScaleName, any, AxisProps>;
+};
+
+export type CartesianContextState = {
+  /**
+   * Mapping from x-axis key to scaling configuration.
+   */
+  xAxis: DefaultizedAxisConfig<ChartsXAxisProps>;
+  /**
+   * Mapping from y-axis key to scaling configuration.
+   */
+  yAxis: DefaultizedAxisConfig<ChartsYAxisProps>;
+  /**
+   * The x-axes IDs sorted by order they got provided.
+   */
+  xAxisIds: AxisId[];
+  /**
+   * The y-axes IDs sorted by order they got provided.
+   */
+  yAxisIds: AxisId[];
+};
+
+export type ZoomData = { axisId: AxisId; start: number; end: number };
+
+export type ZoomFilterMode = 'keep' | 'discard' | 'empty';
+export type ZoomOptions = Record<AxisId, { filterMode: ZoomFilterMode }>;
+
+export type ZoomAxisFilter = (
+  value: number | Date | string | null | { x: number; y: number },
```

Comment:
Why not complitely moving to
```suggestion
  value: { x: number | Date | string | null; y: number | Date | string | null },
```

I see is no reason that a scatter chart could not have string or Date in its axes. The case `string` could be ignored for now. I 've never see someone using the scatter with ordinal scale